rcl/include/rcl/guard_condition.h

```diff
@@ -262,6 +271,34 @@ RCL_WARN_UNUSED
 rmw_guard_condition_t *
 rcl_guard_condition_get_rmw_handle(const rcl_guard_condition_t * guard_condition);
 
+/// Set the on trigger callback function for the guard_condition.
+/**
+ * This API sets the callback function to be called whenever the
+ * guard_condition is triggered.
+ *
```

Comment:
I think it would be nice to add explanation about how `trigger_count` works internally.